docker/reference/reference.go

```diff
@@ -5,7 +5,9 @@
 //
 //	reference                       := name [ ":" tag ] [ "@" digest ]
 //	name                            := [domain '/'] path-component ['/' path-component]*
-//	domain                          := domain-component ['.' domain-component]* [':' port-number]
+//	domain                          := host [':' port-number]
+//	host                            := domain-name | IPv4address | \[ IPv6address \]	; rfc3986 appendix-A
```

Comment:
It’s _not at all_ obvious that we want this. c/image/storage uses a `[storageOptions]name` syntax already, how do we disambiguate?

Reply:
No idea, I was just force updating, I have no problem removing the host definition from this.

Reply:
… and then we need to somehow manage the divergence from the upstream package, if one emerges.

I’m fine with that happening in principle, I just can’t prioritize this.

Reply:
A possible path forward: Update not to the head of the branch, but to some specific commit that contains exactly what we want, but nothing that we would have to think about. That way we would get the improvements we want, we would stay in sync, and we could still bring in the other changes when there’s more capacity.

_If_ such a commit exists.

Alternatively, get this reviewed by someone else who can investigate in enough detail.

Reply:
That is an effective fork, selectively choosing features… It’s an option, sure, but it turns the cost of making a decision on those other changes into technical debt to be paid later. When will that “later” ever happen?